toonz/sources/toonz/sceneviewerevents.cpp

```diff
@@ -944,11 +954,15 @@ void SceneViewer::touchEvent(QTouchEvent *e, int type) {
     m_firstPanPoint = e->touchPoints().at(0).pos();
     m_undoPoint     = m_firstPanPoint;
   }
-  if (e->touchPoints().count() == 1 && m_touchActive) {
+  // touchpads must have 2 finger panning for tools and navigation to be functional
+  // on other devices, 1 finger panning is preferred
+  if ((e->touchPoints().count() == 2 && m_touchActive && QTouchDevice::TouchPad)
+      || (e->touchPoints().count() == 1 && m_touchActive && QTouchDevice::TouchScreen)) {
```

Comment:
You cannot use `QTouchDevice::TouchScreen` or `QTouchDevice::TouchPad` directly in the if statement for identifying the device type.
Please see the [Qt reference](http://doc.qt.io/qt-5/qtouchdevice.html#DeviceType-enum) - they are enum with values 0 (for TouchScreen) and 1 (for TouchPad) . 
To identify which device is used, you need to call the function `QTouchDevice::type()` ([reference](http://doc.qt.io/qt-5/qtouchdevice.html#type)) and compare the return value with these enums.

`QTouchDevice::type()` can be called only in the function `SceneViewer::touchEvent(QTouchEvent *e, int type)` like as follows:

```
if (e->device()->type() == QTouchDevice::TouchScreen) {
  // touch screen case
}
else {    // e->device()->type() == QTouchDevice::TouchPad
 // touch pad case
}
```

If you need to know the current device type outside of the `SceneViewer::touchEvent()` , the standard technique would be to keep the `QTouchDevice::DeviceType` value as the member variable of `SceneViewer` class.

I'll try to modify it and post another PR based on yours. Thanks!